### DISCUSSION TITLE
Run async computation in an independent scope

### DISCUSSION THREAD

**Discussion by aboisvert** [lampepfl/gears#107] [General]
Hi there,
Occasionally I'd like to run a computation in an independent scope, effectively "breaking out" of the structured concurrency paradigm.   With ZIO, for example, I'd use `forkDaemon`.

What is the equivalent in gears? If any.

Also, while I'm at it, are there plans to provide a more general scheduler?  I'm thinking something similar to the Java platform `SchedulerExecutorService`, but would provide better Scala/Gears ergonomics.

* https://zio.dev/reference/fiber/fiber.md/#fork-in-global-scope-daemon
* https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html

**Comment by natsukagami** [comment]
Hi! In gears you would unlink a Future to do that.

As for a general scheduler, I don’t think we will try to ship a self made
scheduler soon, but any scheduler implementing Gear’s Scheduler interface
can be used.

Le lun. 18 nov. 2024 à 15:48, Alex Boisvert ***@***.***> a
écrit :

**Comment by m8nmueller** [comment]
Another option, especially if you have multiple tasks to break out, is the spawning Resource. Using `Resource.spawning`, you get a scope like from `Async.group`. But instead of running it in a structured way directly, execution is deferred and wrapped in a `Resource`. You can pass it around, maybe to a top-level where it is possible to run it in a structured way (`Resource.use`). But you can also start it unstructured-ly, obtaining a close-handler (`Resource.allocated`).

Note that this is currently on the main branch only, but not yet released.

**Comment by aboisvert** [comment]
Thank you for the responses!  I'll look into those options and explore what's best for our use-case.
